src/main/java/org/springframework/data/couchbase/core/ExecutableFindByQueryOperation.java

```diff
@@ -117,6 +117,14 @@
 		 */
 		TerminatingFindByQuery<T> matching(Query query);
 
+		/**
+		 * Set the filter for the query to be used.
+		 *
+		 * @param queryString must not be {@literal null}.
+		 * @throws IllegalArgumentException if query is {@literal null}.
+		 */
+		TerminatingFindByQuery<T> raw(String queryString);
```

Comment:
If I understood @mp911de  correctly I think he was referring to that we should rather use `Query.raw(queryString)` instead and omit this overload. So you can do `matching(Query.raw("select ..."))`.

Reply:
Ideally, if it's a string query, then `new StringQuery(…)` would be a decent way (where `StringQuery extends Query`)

Reply:
Ok, got it. I misunderstood.

Reply:
This still needs to be removed, right?

Reply:
You're right.  I removed the usage but left the definiton.  Amend coming...